\documentclass[poster, final]{jobim}
% Available options are:
% - showframe 
% - draft
% - final [default]

\usepackage[utf8]{inputenc}

% WARNING: already loaded packages:
% - hyperref
% - times
% - color
% - xspace
% - graphicx
% - fancyhdr
% - fancybox
% - indentfirst
% - geometry
% - babel (options english,francais) :
%   choose the language with \selectlanguage{<language>}
\usepackage[
style=numeric,
datamodel=software, % extend the datamodel with entries for software
abbreviate=false,
natbib=true,
sorting=none,
backend=biber,
bibencoding=utf8,
giveninits=true,
url=false,
doi=false,
defernumbers,
maxcitenames=10,
defernumbers=true,
maxbibnames=100]{biblatex}

\addbibresource{jobim_2021_proceedings_SWH.bib}

\pagestyle{empty}
\addtolength{\parskip}{0.4\baselineskip}

%% Title of the paper (required)
\title{Reproducibility in bioinformatics: take care of your code with Software Heritage}

%% List of authors (separated by the macro \and).
%% Authors can be followed by \inst{<n>} macro.
%% The <n> parameter of the \inst macro should correspond to the <n>th institution
%% (see macro \institute below).
\author{Pierre \textsc{Poulain}\inst{1} \and Morane \textsc{Gruenpeter}\inst{2} \and Roberto \textsc{Di Cosmo}\inst{3}}

%% List of institutions (separated by the macro \and).
\institute{
	Université de Paris, CNRS, Institut Jacques Monod, F-75006, Paris, France
	\and
	Software Heritage, Inria, France
	\and
	Software Heritage, Inria and University of Paris, France
}

% email of the corresponding author
\corresponding{pierre.poulain@u-paris.fr}

\begin{document}

% Si vous écrivez en français, commentez la ligne suivante
\selectlanguage{english}
% Si vous écrivez en francais, décommentez la ligne suivante...
% \selectlanguage{francais}

\maketitle

\setcounter{footnote}{0}

Reproducibility is an ongoing effort in the bioinformatics community\cite{kim2018}. Open science helps toward this goal with open access to the scientific literature, open data and
open source research software. In 2018, more than 36\% of yearly published papers were published under open access conditions\cite{trendsopenaccesspublications}. In biology and bioinformatics, the recent development of preprints has acted as a leverage towards open access.

Raw data deposit in public international repositories of genomics and proteomics data is now well established and enforced by most journal editorial policies. Availability of all-purpose data repositories such as Zenodo or Figshare also fostered open data.

It is now important to establish good practices also for scientific software,
going beyond the common approach of depositing code in development platforms
such as GitHub or GitLab, where long-term preservation is not guaranteed.

This presentation aims to present to our community the Software Heritage (SWH)\footnote{\url{https://www.softwareheritage.org/}} archive\cite{dicosmo2017}: it collects, preserves, and makes available all source codes, from the one that ran on the Apollo 11 Guidance Computer to the source code of the Gromacs molecular dynamics engine, the Bowtie 2 genomics read aligner, the Cytoscape network visualization software... Software Heritage can also archives smaller programs like the scripts commonly used in bioinformatics.

Software Heritage regularly collects source code from a growing list of code
hosting platforms, and provides a powerful “Save code now” functionality\footnote{\url{https://archive.softwareheritage.org/save/}} that allows to trigger archival for any
public repository based on the Git, Mercurial or Subversion version control systems, free of charge. Any object archived in Software Heritage is assigned an intrinsic persistent identifier\footnote{\url{https://www.softwareheritage.org/2020/07/09/intrinsic-vs-extrinsic-identifiers/}} called the SWHID\cite{dicosmo2020},
that can be independently verified.

We will present actionable recommendations for better referencing and indexing research source code, including best practices for providing metadata files in the code repository (AUTHOR(s) file with the list of authors, LICENSE file with the applicable license to the source code, README file with the description of the software and other valuable information) and for making it citable\footnote{\url{https://codemeta.github.io/}}, including pointers to appropriate bibliographic styles\cite{dicosmo2020}.


\printbibliography
 
\end{document}

